tests/agents/gherkin-specs/azure_functions_metadata.feature

```diff
@@ -0,0 +1,70 @@
+      | WEBSITE_OWNER_NAME          | d2cd53b3-acdc-4964-9563-3f5201556a81+faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME           | site_name                                                               |
+    When cloud metadata is collected
+    Then cloud metadata is not null
+    And cloud metadata 'account.id' is 'd2cd53b3-acdc-4964-9563-3f5201556a81'
+    And cloud metadata 'provider' is 'azure'
+    And cloud metadata 'service.name' is 'functions'
+    And cloud metadata 'instance.name' is 'site_name'
+    And cloud metadata 'project.name' is 'faas_group'
+    And cloud metadata 'region' is 'CentralUS'
+
+  Scenario Outline: Azure Function App with typical set of environment variables present in expected format
+    Given the following environment variables are present
+      | name                        | value                                                                   |
+      | FUNCTIONS_EXTENSION_VERSION | version                                                                 |
+      | WEBSITE_OWNER_NAME          | d2cd53b3-acdc-4964-9563-3f5201556a81+faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME           | site_name                                                               |
+      | REGION_NAME                 | Central US                                                              |
+      | WEBSITE_RESOURCE_GROUP      | faas_group_from_env                                                     |
+    When cloud metadata is collected
+    Then cloud metadata is not null
+    And cloud metadata 'account.id' is 'd2cd53b3-acdc-4964-9563-3f5201556a81'
+    And cloud metadata 'provider' is 'azure'
+    And cloud metadata 'service.name' is 'functions'
+    And cloud metadata 'instance.name' is 'site_name'
+    And cloud metadata 'project.name' is 'faas_group_from_env'
+    And cloud metadata 'region' is 'Central US'
+
+  Scenario: WEBSITE_OWNER_NAME environment variable not expected format
+    Given the following environment variables are present
+      | name               | value                                                                   |
+      | WEBSITE_OWNER_NAME | d2cd53b3-acdc-4964-9563-3f5201556a81-faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME  | site_name                                                               |
+    When cloud metadata is collected
+    Then cloud metadata is null
+
+  Scenario: Missing FUNCTIONS_EXTENSION_VERSION environment variable
+    Given the following environment variables are present
+      | name               | value                                                                   |
+      | WEBSITE_OWNER_NAME | d2cd53b3-acdc-4964-9563-3f5201556a81+faas_group-CentralUSwebspace-Linux |
+      | WEBSITE_SITE_NAME  | site_name                                                               |
+    When cloud metadata is collected
+    Then cloud metadata is null

```

Comment:
The metadata section of the Azure Functions spec (https://github.com/elastic/apm/blob/main/specs/agents/tracing-instrumentation-azure-functions.md#metadata) says that `FUNCTIONS_WORKER_RUNTIME` should be used to decide if we are in an Azure Function app environment, not `FUNCTIONS_EXTENSION_VERSION`.

Should we adjust these gherkin scenarios accordingly?  Currently the Node.js agent instrumentation gates on the presence of `FUNCTIONS_WORKER_RUNTIME`.   (It uses `FUNCTIONS_EXTENSION_VERSION` for `metadata.service.framework.runtime.version`, but doesn't disable metadata collection if it isn't present.)

Reply:
Right, there is an inconsistency.
The reason I chose `FUNCTIONS_EXTENSION_VERSION` for my tests is that this variable was vital. Removing it causes the runtime not to come up at all.
But testing for `FUNCTIONS_WORKER_RUNTIME` is just as fine.

Reply:
> Removing it causes the runtime not to come up at all.

You mean when using the `func start` tooling for locally running?

Reply:
No, in actual Azure.
Removing `FUNCTIONS_EXTENSION_VERSION` brings down the function app (HTTP 503).
Removing `FUNCTIONS_WORKER_RUNTIME` technically still works for .NET (and falls back to `dotnet-isolated`). However, I would still consider this unsupported and instrumentations should probably check for this variable.

Reply:
Ah. Thanks for the details.